bundles/mapping/layerswipe/instance.js

```diff
@@ -71,14 +72,26 @@ Oskari.clazz.define(
                 this.unregisterEventListeners();
                 this.hideSplitter();
                 Oskari.getSandbox().getService('Oskari.mapframework.service.VectorFeatureService').setHoverEnabled(true);
+                this.setSwipeStatus(null, false, null);
             }
             this.active = active;
             this.mapModule.getMap().render();
         },
 
+        setSwipeStatus: function (layerId, active, cropX) {
```

Comment:
Move layer variable setting here:
```suggestion
        setSwipeStatus: function (layerId, active, cropX) {
            this.oskariLayerId = layerId;
```